meson/arch/amd64/meson.build

```diff
@@ -0,0 +1,46 @@
+# TODO: Red Zone is only a problem if we allow interrupts to land on an active
```

Comment:
So what's wrong with the argument?

In general, I'd suggest to put ideas for improvements into the bug tracker instead of forgetting them forever somewhere in the build files.

Reply:
Fair point. I'll move those to tickets.

Reply:
As for the `-mno-red-zone` argument, I consider it a hack. It's just another way kernel code needs to be special. IMO the "proper" solution is to set up interrupt stacks in such a way that every interrupt always lands on a clean stack. Apart from generating better code, it's easier to make sure kernel stacks never overflow that way.